orchagent/orch.cpp

```diff
@@ -188,14 +189,22 @@ size_t Consumer::refillToSync()
         } while (update_size != 0);
         return total_size;
     }
-    else
+    else if (consumertable != NULL)
     {
         // consumerTable is either ConsumerStateTable or ConsumerTable
-        auto db = getDbConnector();
+        auto db = consumertable->getDbConnector();
         string tableName = getTableName();
         auto table = Table(db, tableName);
         return refillToSync(&table);
     }
+    else
+    {
+        // The consumerTable should be ZmqConsumerStateTable.
+        DBConnector db("APPL_DB", 0);
```

Comment:
Please check if the consumerTable is ZmqConsumerStateTable and use getDbConnector() in ZmqConsumerStateTable:

https://github.com/sonic-net/sonic-swss-common/blob/5b6377cd260e5421470263ebd7e27ac8aca42c52/common/zmqconsumerstatetable.h

The reason of this is because the APPL_DB may not in local and may not be 'APPL_DB', the design still under discussion, so if use DBConnector from ZMQ, the code can keep no change when design finalized.

Reply:
Done